### ISSUE TITLE
Change part scheduling according to shipments instead of sales orders

### ISSUE THREAD

**Issue by SchrodingersGat** [inventree/InvenTree#5181]
### Discussed in https://github.com/inventree/InvenTree/discussions/5180

<div type='discussions-op-text'>

<sup>Originally posted by **HannesMorre** July  5, 2023</sup>
Since it is possible to split a sales order into multiple shipments, it seems logical to me that part scheduling should be adjusted according to these shipments instead of sales orders. See the example below:

There is a sales order with target date 30/08/2023. This sales order holds 3 line items (articles A, B and C) each 50 pcs.
Articles A and B are fully allocated to shipment 1 (which is scheduled for 15/07/2023)
25 pcs of article C are allocated to shipment 2 (scheduled for 01/08/2023)
25 pcs of article C are allocated to shipment 3 (scheduled for 15/08/2023)

In the part scheduling of these parts (A, B and C) we will see a drop on 30/08/2023. I think this should be 15/07/2023 for articles A and B, and for article C this drop should be split up on 01/08/2023 and 15/08/2023 instead of 30/08/2023.

Or am I doing something wrong?</div>

**Comment by miggland** [comment]
Which date are you using to schedule the shipment?

There are two dates:
- shipment date
- delivery date

The shipment date is intended to mark when a shipment is actually sent out. The delivery date has no logic connected to it, but was intended to be set when the shipment had been delivered. It could be used as a target date.

**Comment by matmair** [comment]
For the internal projections, it makes sense to use shipment day IMO - as the stock is out of the house from there on.

**Comment by miggland** [comment]
It does - but that date is filled on shipping at the moment, and isn't directly available to set in advance. It if is set, the shipment is marked as shipped.

**Comment by SchrodingersGat** [comment]
For this to work, we might have to add a "target date" to outgoing shipments? This is only really applicable when you have projected shipment dates

**Comment by miggland** [comment]
Yes, that would make it clean, and would be better than using delivery date.

**Comment by HannesMorre** [comment]
When is this planned to be implemented?

**Comment by matmair** [comment]
@HannesMorre  There is no plan to implement this, if someone picks it up it will be implemented. If not it will be closed at some point.
Things that are (somewhat) on the roadmap have either the label `roadmap` label or someone is assigned (which means the person is interested in implementing it). We do not provide implementation or release timelines.

**Comment by HannesMorre** [comment]
Okay, thanks for the info. This means there is a chance this will never be implemented, right?

**Comment by wolflu05** [comment]
Possibly yes, but you can develop this yourself and propose a PR or maybe also sponsor this development if you really need it. But details you would need to discuss with @SchrodingersGat

**Comment by wolflu05** [comment]
Possibly yes, but you can develop this yourself and propose a PR or maybe also sponsor this development if you really need it. But details you would need to discuss with @SchrodingersGat

**Comment by HannesMorre** [comment]
I am not currently active as a developer, but looking into starting over again so I can contribute to the project in my free time.

**Comment by SchrodingersGat** [comment]
Not stale

**Comment by SchrodingersGat** [comment]
Not stale

**Comment by HannesMorre** [comment]
not stale

**Comment by bberg115** [comment]
I have a similar request for part scheduling. 

Additionally, we would like some options for filtering and what is presented.
1. Having a filter to only use issued sales orders in the view. We use Sales Orders for quotes and actual sales orders. Ideally, the "issued" state could be use as some sort of filter.
2. Total count - it's shown on the graph, can the value be shown in the interace somewhere.
3. Downloadable - can the data in the table be downloadable (csv)?
